### ISSUE TITLE
Async methods are treated as falsy in template conditionals

### ISSUE THREAD

**Issue by DuBistKomisch** [emberjs/ember.js#18641]
```
export default class Whatever extends Component {
  doSomething() {
    ...
  }

  async doSomethingAsync() {
    ...
  }
}
```

```
{{if this.doSomething 'true' 'false'}}
{{if this.doSomethingAsync 'true' 'false'}}
```

This renders `true false`, but should render `true true`. Happens in latest Firefox and Chromium on Linux. I'm not using the babel polyfill AFAIK, and I'm on latest stable Ember (3.15.0) but I've seen this on previous versions without realising what was happening.

This is particularly problematic when passing optional `@action`s into other components. Workaround is to wrap the async method in a sync method and use that instead.

**Comment by pzuraq** [comment]
This does seem like a bug, I would have expected an `async` method to still be truthy. We should look into this asap.

**Comment by rwjblue** [comment]
Fixed by #18651
